lib/analysis_server.dart

```diff
@@ -546,14 +547,27 @@ class _AnalysisServerWrapper extends Server {
     }
     List<String> arguments = [path, '--sdk=${sdk.path}'];
 
-    if (AnalysisServer.startWithDebugging) {
+    // Check to see if we should start with diagnostics enabled.
+    if (AnalysisServer.startWithDiagnostics) {
       arguments.add('--port=${AnalysisServer.DIAGNOSTICS_PORT}');
       _logger.info('analysis server diagnostics available at '
           '${AnalysisServer.diagnosticsUrl}.');
     }
 
+    // Start in checked mode?
     if (AnalysisServer.useChecked) {
-      arguments.insert(0, '--checked');
+      arguments.add('--checked');
```

Comment:
I think this change is wrong. The `--checked` is a VM argument, so it really does need to come before the other arguments. (Similar issue in entry.dart.js.)